Include/unionobject.h

```diff
@@ -0,0 +1,13 @@
+#ifndef Py_UNIONTYPEOBJECT_H
+#define Py_UNIONTYPEOBJECT_H
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+PyAPI_FUNC(PyObject *) Py_Union(PyObject *);
```

Comment:
@gvanrossum This functions need to be documented in the C-API, no? If this is not intended to be public, then we should add an underscore to the name

Reply:
I'm not sure I see a use case for extension modules constructing unions yet, so I am fine with making it private. We can always change our mind later -- not so easily the other way around.

Reply:
Ok, 47a4cb3 makes the API private for the time being.